server/app/utils.py

```diff
@@ -24,4 +24,14 @@ def first(condition: Callable[[T], bool], items: Iterable[T]) -> Optional[T]:
     for item in items:
         if condition(item):
             return item
-    return None
\ No newline at end of file
+    return None
+
+def get_footer():
+    """Generates page footer. Tries to load timestamp.txt (which is generated by update.sh script)"""
+    version = "unknown"
+    try:
+        f = open("timestamp.txt", 'r')
+        version = f.read()
+        return "Version <a href='https://github.com/gut-space/satnogs/commit/%s'>%s</a>" % (version, version)
```

Comment:
You should avoid create HTML in controller. Your controller should return pure data. In this case it should return ```version```. If file doesn't exist it should return ```version``` equals ```None```. In Jinja you can handle it using ```{% if version %}```